### PR TITLE
grpc_json_transcoder: to support reject request if binding and body value are conflict

### PR DESCRIPTION
**Pull request by shuoyang2016**
Commit Message:
Updated `grpc-httpjson-transcoding` repo to have this [change](grpc-ecosystem/grpc-httpjson-transcoding#70).
Additional Description:
grpc_json_transcoder: to support reject request if binding and body value are conflict
Risk Level: Low
Testing: unit test
Docs Changes: N/A
Release Notes: Add an option in grpc_json_transcoder to support reject request if binding and body value are conflict.
Platform Specific Features: N/A

### CODE REVIEW

File: test/extensions/filters/http/grpc_json_transcoder/json_transcoder_filter_test.cc [hunk 1]
```diff
@@ -1321,6 +1321,40 @@ TEST_F(GrpcJsonTranscoderFilterTest, TranscodingStreamWithFragmentedHttpBody) {
...
+
+  EXPECT_CALL(decoder_callbacks_, clearRouteCache());
+
+  EXPECT_EQ(Http::FilterHeadersStatus::Continue, filter_.decodeHeaders(request_headers, false));
+  EXPECT_EQ("application/grpc", request_headers.get_("content-type"));
+  EXPECT_EQ("/shelf/123", request_headers.get_("x-envoy-original-path"));
+  EXPECT_EQ("POST", request_headers.get_("x-envoy-original-method"));
+  EXPECT_EQ("/bookstore.Bookstore/CreateShelfBodyWildcard", request_headers.get_(":path"));
+  EXPECT_EQ("trailers", request_headers.get_("te"));
+  EXPECT_CALL(decoder_callbacks_, sendLocalReply(Http::Code::BadRequest, _, _, _, _));
```
Comment (qiwzhang) [inline test/extensions/filters/http/grpc_json_transcoder/json_transcoder_filter_test.cc:1350]: line 1345, 1349 are not important, and it cause distracting,  suggest to remove

add comment for line 1350 as:

//  decodeData() will cause request to be rejected due to ...

**Review by qiwzhang — DISMISSED** [review]
A minor suggestion

File: test/extensions/filters/http/grpc_json_transcoder/json_transcoder_filter_test.cc (hunk 1 above)
Comment (shuoyang2016) [reply test/extensions/filters/http/grpc_json_transcoder/json_transcoder_filter_test.cc:1350]: Thanks. Done.

**Comment by shuoyang2016** [comment]
friendly ping

**Review by lizan — APPROVED** [review]

**Pull request merged**